anndata/core/anndata.py

```diff
@@ -1420,9 +1420,8 @@ def _remove_unused_categories(self, df_full, df_sub, uns):
             if color_key not in uns:
                 continue
             color_vec = uns[color_key]
-            # this is a strange hack...
-            # What case is this? Does this happen?
             if np.array(color_vec).ndim == 0:
+                # Make 0D arrays into 1D ones
```

Comment:
Do we want this to happen?

Reply:
I would assume not. If it's not a one dimensional array, I don't think we put it there.

Reply:
@falexwolf implemented that specifically in ac08c617458dc500bc5a308f368bb28c132d5d26.

Alex, can you help here?